Makefile

```diff
@@ -18,8 +18,19 @@ bootstrap: virtualenv
 requirements: virtualenv requirements.txt
 	${VIRTUALENV_ROOT}/bin/pip install -r requirements.txt
 
-requirements_for_test: virtualenv requirements_for_test.txt
-	${VIRTUALENV_ROOT}/bin/pip install -r requirements_for_test.txt
+requirements-dev: virtualenv requirements-dev.txt
+	${VIRTUALENV_ROOT}/bin/pip install -r requirements-dev.txt
+
+compile-requirements:
+	virtualenv venv-freeze
+	$$(pwd)/venv-freeze/bin/pip install -r requirements-app.txt
+	echo '# This file is autogenerated. Run `make compile-requirements`' > requirements.txt
+	echo '# to update it with any changes made in requirements-app.txt' >> requirements.txt
+	echo '' >> requirements.txt
+	cat requirements-app.txt >> requirements.txt
+	echo '' >> requirements.txt
+	$$(pwd)/venv-freeze/bin/pip freeze -r requirements-app.txt | sed -n '/The following requirements were added by pip freeze/,$$p' >> requirements.txt
+	rm -rf venv-freeze
```

Comment:
It's probably best to do any `rm -rf venv-freeze` at the beginning of the process to clean up the results of any failed runs.